src/snarks/groth16/evaluation_from_lagrange.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "include_libsnark.hpp"
+
+#include <map>
+
+/// Given a sequence of powers of x (with some factor) encoded in
+/// GroupT, compute the values of various linear combination of
+/// Lagrange polynomials at x.
+///
+/// TODO: Here, we assume that there may be more Lagrange polynomails
+/// than variables in the original qap, so we use a map to avoid
+/// evaluating those that are unused.  This may not be the optimal approach.
+template<typename ppT, typename GroupT> class evaluation_from_lagrange
+{
+public:
+    evaluation_from_lagrange(
+        const std::vector<GroupT> &powers,
+        libfqfft::evaluation_domain<libff::Fr<ppT>> &domain);
+
+    GroupT evaluate_from_langrange_factors(
```

Comment:
`evaluate_from_langrange_factors` -> `evaluate_from_lagrange_factors`

Reply:
Typo should be fixed in latest.  Thanks.